### PR TITLE
Add Regulation and Requisite Registration

### LINKED ISSUE #74: Regulation Crud

### PR DESCRIPTION
**Pull request by roalcantara**
closes #74

### CODE REVIEW

**Comment by roalcantara** [comment]
Hey @ddomingues,
Can you check this PR out?
Cheers.

File: db/migrate/20161109005503_add_item_and_title_to_requisites.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class AddItemAndTitleToRequisites < ActiveRecord::Migration[5.0]
+  def change
+    add_column :requisites, :item, :string, null: false
+    add_column :requisites, :title, :string
+    add_index :requisites, [:item, :parent_id], unique: true
+    add_index :requisites, [:item, :regulation_id], unique: true
+    change_column_null :requisites, :description, true
```
Comment (roalcantara) [inline db/migrate/20161109005503_add_item_and_title_to_requisites.rb]: Hey @ddomingues, what about this?

File: app/controllers/admin/regulations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  def regulation_params
+    params.require(:regulation).permit(:name, :description)
+  end
+
+  def add_breadcrumbs
+    add_breadcrumb Regulation
+    if @regulation
+      add_breadcrumb @regulation.to_s
+    else
+      add_breadcrumb Regulation.model_name.human
```
Comment (ddomingues) [inline app/controllers/admin/regulations_controller.rb]: What about?

`add_breadcrumb (@regulation&.to_s || Regulation.model_name.human)`

File: app/assets/stylesheets/overrides/plugins/nestable.scss [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    padding-left: 15px;
+}
+
+.dd-item-alt {
+  .dd-content {
+    display: block;
+    height: 34px;
+    -webkit-box-sizing: border-box;
+    -moz-box-sizing: border-box;
+    box-sizing: border-box;
```
Comment (ddomingues) [inline app/assets/stylesheets/overrides/plugins/nestable.scss]: It isn't a good approach to use browser vendor prefix...

File: app/controllers/admin/requisites_controller.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    parameters
+  end
+
+  def add_breadcrumbs
+    add_breadcrumb Regulation
+    add_breadcrumb @regulation
+    if @requisite
+      add_breadcrumb @requisite.to_s
+    else
+      add_breadcrumb Requisite.model_name.human
```
Comment (ddomingues) [inline app/controllers/admin/requisites_controller.rb]: `add_breadcrumb @requisite&.to_s || Regulation.model_name.human`?

File: spec/routing/admin/requisites_routing_spec.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
+require "rails_helper"
+
+RSpec.describe Admin::RequisitesController do
+  mock_warden_for_route_tests_with :admin
+
+  describe "routing" do
+    it "routes to #new" do
+      expect(get: "/admin/regulations/1/requisites/new").
+        to route_to("admin/requisites#new", regulation_id: "1")
+    end
```
Comment (ddomingues) [inline spec/routing/admin/requisites_routing_spec.rb]: @roalcantara you forgot of add a routing test to `get "/:parent_id/new", to: "requisites#new", as: :add_requisite`? right?

File: app/views/admin/tasks/_requisites.html.slim [hunk 6]
```diff
@@ -0,0 +1,15 @@
+ol.dd-list
+  li.dd-item.dd-item-alt data-id="#{requisite.id}"
+    .dd-content class="#{"has-children" if requisite.children.any?}"
+      - unless requisite.description.nil?
+        span.pull-right.table-actions
+          a href="#" title="#{requisite.description}"
+            i.icon.zmdi.zmdi-help-outline
+      span.checkbox.checkbox-custom.checkbox-primary.margin-0.margin-left-5.margin-right-15
+        = check_box_tag "task[requisite_ids][]", requisite.id, @task.requisites.include?(requisite),
```
Comment (ddomingues) [inline app/views/admin/tasks/_requisites.html.slim]: I will think about it after this merge. Sorry 🤗

**Pull request merged**